glustercli/cmd/common.go

```diff
@@ -54,35 +54,36 @@ func handleGlusterdConnectFailure(msg, endpoints string, err error, errcode int)
 
 func failure(msg string, err error, errcode int) {
 
-	handleGlusterdConnectFailure(msg, GlobalFlag.Endpoints[0], err, errcode)
-
 	w := os.Stderr
+	printErr := func(err error) {
+		if err != nil {
+			fmt.Fprintln(w, err)
+		}
+	}
+
+	handleGlusterdConnectFailure(msg, GlobalFlag.Endpoints[0], err, errcode)
 
-	w.WriteString(msg + "\n")
+	fmt.Fprintln(w, msg)
 
-	if client == nil && err != nil {
-		fmt.Fprintln(w, err)
+	gd2client, ok := client.(*restclient.Client)
+	if client == nil || !ok {
+		printErr(err)
 		os.Exit(errcode)
 	}
 
-	resp := client.LastErrorResponse()
-
-	if resp == nil && err != nil {
-		fmt.Fprintln(w, err)
+	resp := gd2client.LastErrorResponse()
+	if resp == nil {
+		printErr(err)
 		os.Exit(errcode)
 	}
 
-	if err != nil {
-		w.WriteString("\nResponse headers:\n")
-		for k, v := range resp.Header {
-			if strings.HasSuffix(k, "-Id") {
-				w.WriteString(fmt.Sprintf("%s: %s\n", k, v[0]))
-			}
+	fmt.Fprintln(w, "\nResponse headers:")
+	for k, v := range resp.Header {
+		if strings.HasSuffix(k, "-Id") {
+			fmt.Fprintf(w, "%s: %s\n", k, v[0])
```

Comment:
as a safer side can we add a check for length of v. accessing index may cause panic